lib/Dialect/Torch/Transforms/AdjustCallingConventions.cpp

```diff
@@ -187,8 +188,49 @@ class AdjustCallingConventionForReturn
 };
 } // namespace
 
+static bool isValidNonContainerResultType(Type result_type) {
+  if (result_type.isa<Torch::BaseTensorType>() ||
+      result_type.isa<Torch::FloatType>() ||
+      result_type.isa<Torch::IntType>() ||
+      result_type.isa<Torch::BoolType>()) {
+    return true;
+  }
+  return false;
+}
+
+static LogicalResult validateReturns(func::FuncOp func) {
+  // getResultTypes can only ever be one item.  Multiple return values are
+  // stored in a tuple.
```

Comment:
This is true in our pipeline, but you could still hit the assert with verifier-valid IR. `assert` should only be used for scenarios where verifier-valid IR could never hit it.

I do think we should emit an error diagnostic in this case though, since it should hold for our pass pipelines.